libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -154,6 +154,15 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 	}
 }
 
+
+
+string IRGeneratorForStatements::evaluateExpression(Expression const& _expression, Type const& _targetType)
+{
+	IRVariable variable{_expression};
+	m_code << "let " + variable.name() + " := " + expressionAsType(_expression, _targetType) + "\n";
+	return variable.name();
+}
```

Comment:
Maybe something like:
```suggestion
IRVariable IRGeneratorForStatements::evaluateExpression(Expression const& _expression, Type const& _targetType)
{
	_expression.accept(*this); // The expression wasn't visited before, was it? Then you need to do it here or on the caller side - if it was already visited, you can remove this.
	IRVariable variable{m_context.newYulVariable(), _targetType};
	variable.define(_expression);
	return variable;
}
```
And then just use ``evaluateExpression(...).commaSeparatedList()`` on the caller side?